templates/15_insert.go.tpl

```diff
@@ -201,3 +201,168 @@ CacheNoHooks:
+			return err
+		}
+
+		{{- if not .NoAutoTimestamps -}}
+		{{- $colNames := .Table.Columns | columnNames -}}
+		{{if containsAny $colNames "created_at" "updated_at"}}
+		currTime := time.Now().In(boil.GetLocation())
+		{{range $ind, $col := .Table.Columns}}
+		{{- if eq $col.Name "created_at" -}}
+		{{- if eq $col.Type "time.Time" }}
+		if row.CreatedAt.IsZero() {
+			row.CreatedAt = currTime
+		}
+		{{- else}}
+		if queries.MustTime(row.CreatedAt).IsZero() {
+			queries.SetScanner(&row.CreatedAt, currTime)
+		}
+		{{- end -}}
+		{{- end -}}
+		{{- if eq $col.Name "updated_at" -}}
+		{{- if eq $col.Type "time.Time"}}
+		if row.UpdatedAt.IsZero() {
+			row.UpdatedAt = currTime
+		}
+		{{- else}}
+		if queries.MustTime(row.UpdatedAt).IsZero() {
+			queries.SetScanner(&row.UpdatedAt, currTime)
+		}
+		{{- end -}}
+		{{- end -}}
+		{{end}}
+		{{end}}
+		{{- end}}
+
+		nzDefaults := queries.NonZeroDefaultSet({{$alias.DownSingular}}ColumnsWithDefault, row)
+
+		wl, returnColumns := columns.InsertColumnSet(
+			{{$alias.DownSingular}}Columns,
+			{{$alias.DownSingular}}ColumnsWithDefault,
+			{{$alias.DownSingular}}ColumnsWithoutDefault,
+			nzDefaults,
+		)
+
+		if i == 0 {
+			sql = "INSERT INTO {{$schemaTable}} " + "({{.LQ}}" + strings.Join(wl, "{{.RQ}},{{.LQ}}") + "{{.RQ}})" + " VALUES "
+			{{if not .Dialect.UseLastInsertID -}}
+			retQuery = " RETURNING {{.LQ}}" + strings.Join(returnColumns, "{{.RQ}},{{.LQ}}") + "{{.RQ}} "
+			{{end -}}
+		}
+
+		sql += strmangle.Placeholders(dialect.UseIndexPlaceholders, len(wl), len(vals)+1, len(wl))
+		if i != len(o)-1 {
+			sql += ","
+		}
+
+		valMapping, err := queries.BindMapping({{$alias.DownSingular}}Type, {{$alias.DownSingular}}Mapping, wl)
+		if err != nil {
+			return err
+		}
+		retMapping, err := queries.BindMapping({{$alias.DownSingular}}Type, {{$alias.DownSingular}}Mapping, returnColumns)
+		if err != nil {
+			return err
+		}
+		value := reflect.Indirect(reflect.ValueOf(row))
+		vals = append(vals, queries.ValuesFromMapping(value, valMapping)...)
+		retMaps = append(retMaps, queries.PtrsFromMapping(value, retMapping)...)
+	}
+
+	{{if not .Dialect.UseLastInsertID -}}
+	sql += retQuery
+	{{end -}}
+
+	if boil.DebugMode {
+		fmt.Fprintln(boil.DebugWriter, sql)
+		fmt.Fprintln(boil.DebugWriter, vals...)
+	}
+
+	{{if .Dialect.UseLastInsertID -}}
+	{{- $canLastInsertID := .Table.CanLastInsertID -}}
+	{{if $canLastInsertID -}}
+		{{if .NoContext -}}
+	result, err := exec.Exec(ctx, sql, vals...)
+		{{else -}}
+	result, err := exec.ExecContext(ctx, sql, vals...)
+		{{end -}}
+	{{else -}}
+		{{if .NoContext -}}
+	_, err := exec.Exec(ctx, sql, vals...)
+		{{else -}}
+	_, err := exec.ExecContext(ctx, sql, vals...)
+		{{end -}}
+	{{- end}}
+	if err != nil {
+		return errors.Wrap(err, "{{.PkgName}}: unable to insert into {{.Table.Name}}")
+	}
+	{{if $canLastInsertID -}}
+	var lastID int64
+	{{- end}}
+
+	{{if $canLastInsertID -}}
+	lastID, err = result.LastInsertId()
```

Comment:
If this code is not looped over as the rest of it is, there's no chance that mysql values will be returned for all inserted rows.